src/components/CloneIssuesButton.jsx

```
@@ -0,0 +1,69 @@
+import React from "react";
+import { useLocation } from "@docusaurus/router";
+import { API_URL, CLIENT_ID } from "../constants";
+
+const SignInButton = () => {
+  const location = useLocation();
+  const login = () => {
+    const state = JSON.stringify({
+      prevPath: `${location.pathname}`,
+    });
+    window.open(
+      `https://github.com/login/oauth/authorize?client_id=${CLIENT_ID}&state=${state}`,
+      "_self"
+    );
+  };
+
+  return <button onClick={login}>Sign In</button>;
+};
+
+const CloneButton = ({ repo, issue = null, token = null }) => {
+  if (!repo || !token) {
+    return null;
+  }
```

This is another case where we're technically using a hook conditionally: it is possible for `useState` to not run.

I think I would throw an error if these props aren't provided.